pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -476,6 +476,16 @@ func getStsAnnotations(val *Values, sts *appsv1.StatefulSet) map[string]string {
 		},
 		val.Annotations,
 	)
+
+	// If `scaleToMultiNodeAnnotationKey` annotation is already present in etcd statefulset
+	// then it is better to check scale-up was successful or not before removing `scaleToMultiNodeAnnotationKey` annotation.
+	if metav1.HasAnnotation(sts.ObjectMeta, scaleToMultiNodeAnnotationKey) {
+		if (sts.Spec.Replicas != nil && sts.Status.UpdatedReplicas < *sts.Spec.Replicas) || sts.Status.Replicas > sts.Status.UpdatedReplicas {
```

Comment:
Should not we also compare with ETCD CR replicas? It would add ultimate safety check

Reply:
I don't want any bug in our controllers can lead to add/remove scale-up annotation in etcd sts.
TBH, right now I would like to depend only on statefulset controller.

Reply:
Okay